### PR TITLE
x86: remove iamcu calling conventions

### PR DESCRIPTION
**Pull request by nashif**
IAMCU was only supported on a few configurations that we are planning to
deprecate.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc changes look OK.

File: arch/x86/CMakeLists.txt [hunk 1]
```diff
@@ -9,24 +9,11 @@ else()
...
-
+set_property(GLOBAL PROPERTY PROPERTY_OUTPUT_ARCH "i386")
+set_property(GLOBAL PROPERTY PROPERTY_OUTPUT_FORMAT "elf32-i386")
 
 
 if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
-  if(CONFIG_X86_IAMCU)
-    zephyr_compile_options(-miamcu)
-  else()
-    zephyr_compile_options(-Qunused-arguments)
```
Comment (gnuless) [inline arch/x86/CMakeLists.txt]: I think we want to keep `else` clause?

File: include/linker/linker-tool-gcc.h [hunk 2]
```diff
@@ -25,13 +25,8 @@
 #elif defined(CONFIG_ARC)
 	OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
 #elif defined(CONFIG_X86)
-	#if  defined(__IAMCU)
-		OUTPUT_FORMAT("elf32-iamcu")
-		OUTPUT_ARCH("iamcu:intel")
-	#else
 		OUTPUT_FORMAT("elf32-i386", "elf32-i386", "elf32-i386")
```
Comment (gnuless) [inline include/linker/linker-tool-gcc.h]: undent

**Review by ghost — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Comment by andrewboie** [comment]
```
FAILED: ccache /opt/sdk/zephyr-sdk-0.10.0/i586-zephyr-elf/bin/i586-zephyr-elf-gcc -DBUILD_VERSION=zephyr-v1.14.0-1438-gc39911de2011 -DENABLE_EXTERNAL_ISR_HANDLING -DKERNEL -DQM_LAKEMONT -DSOC_SERIES=quark_se -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../../drivers -I../../../../../../../subsys/usb -I../../../../../../../kernel/include -I../../../../../../../arch/x86/include -I../../../../../../../include -I../../../../../../../include/drivers -Izephyr/include/generated -I../../../../../../../soc/x86/intel_quark/quark_se -I../../../../../../../soc/x86/intel_quark/include -I../../../../../../../lib/libc/minimal/include -I/home/buildslave/src/github.com/zephyrproject-rtos/modules/hal/qmsi/include -I/home/buildslave/src/github.com/zephyrproject-rtos/modules/hal/qmsi/drivers/include -I/home/buildslave/src/github.com/zephyrproject-rtos/modules/hal/qmsi/soc/quark_se/include -isystem /opt/sdk/zephyr-sdk-0.10.0/i586-zephyr-elf/bin/../lib/gcc/i586-zephyr-elf/8.3.0/include -isystem /opt/sdk/zephyr-sdk-0.10.0/i586-zephyr-elf/bin/../lib/gcc/i586-zephyr-elf/8.3.0/include-fixed -Os -nostdinc -g -imacros /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/out-3nd-pass/arduino_101/samples/subsys/usb/mass/sample.usb.mass_flash/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -Werror -Wno-deprecated-declarations -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/buildslave/src/github.com/zephyrproject-rtos/zephyr=. -ffunction-sections -fdata-sections -mpreferred-stack-boundary=2 -mno-sse -march=lakemont -mtune=lakemont -msoft-float -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/subsys/usb/usb_descriptor.c.obj -MF zephyr/CMakeFiles/zephyr.dir/subsys/usb/usb_descriptor.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/subsys/usb/usb_descriptor.c.obj   -c /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/usb/usb_descriptor.c
In file included from ../../../../../../../include/logging/log.h:11,
                 from ../../../../../../../include/usb/usb_device.h:41,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/usb/usb_descriptor.c:14:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/usb/usb_descriptor.c: In function 'usb_fix_descriptor':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/usb/usb_descriptor.c:412:13: error: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Werror=format=]
     LOG_DBG("Now the wTotalLength is %d",
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      (u8_t *)head - (u8_t *)cfg_descr);
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../../include/logging/log_core.h:236:27: note: in definition of macro '__LOG'
    log_printf_arg_checker(__VA_ARGS__);      \
                           ^~~~~~~~~~~
../../../../../../../include/logging/log.h:77:25: note: in expansion of macro 'Z_LOG'
 #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
                         ^~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/usb/usb_descriptor.c:412:5: note: in expansion of macro 'LOG_DBG'
     LOG_DBG("Now the wTotalLength is %d",
     ^~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Comment by gnuless** [comment]
@nashif @andrewboie Apparently, i586-zephyr-elf and i586-zephyr-elfiamcu disagree on the type of `ptrdiff_t` (`long int` vs `int`, respectively). That's odd, because the difference is purely academic anyway on 32-bit x86.

**Comment by andrewboie** [comment]
Can we please just delete the Quark platforms first, and *then* remove IAMCU?
You're doing this backwards

**Pull request closed without merging**